pkg/process/monitor/process_monitor.go

```diff
@@ -238,23 +238,17 @@ func (pm *ProcessMonitor) Initialize() error {
 	var initErr error
 	pm.initOnce.Do(
 		func() {
-			pm.done = make(chan struct{})
-			pm.initCallbackRunner()
+			if err := pm.initNetlinkProcessEventMonitor(); err != nil {
+				initErr = err
+				return
+			}
```

Comment:
starting the netlink before the main event loop is wrong, as we might fill on channels before we have the chance to setup readers

Reply:
yes but today we don't guarantee init order (callback runners started, mainloop started, then netlink feed the chan)

I rollback these changes as it wasn't the main point on this PR